src/main/java/core/basesyntax/FigureSuplier.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSuplier {
+    private static Random random = new Random();
+
+    public static Figure getRandomFigure() {
+        int figureNumber = random.nextInt(5);
+        ColorSuplier colorSuplier = new ColorSuplier();
+        Random random = new Random();
+        Circle circle = new Circle(colorSuplier.getRandomColor(),random.nextInt());
+        IsoscelesTrapezoid isoscelesTrapezoid = new IsoscelesTrapezoid(
+                colorSuplier.getRandomColor(),random.nextInt(),
+                random.nextInt(), random.nextInt());
+        Rectangle rectangle = new Rectangle(colorSuplier.getRandomColor(),random.nextInt(),
+                random.nextInt());
+        RightTriangle rightTriangle = new RightTriangle(colorSuplier.getRandomColor(),
+                random.nextInt(), random.nextInt());
+        Square square = new Square(colorSuplier.getRandomColor(), random.nextInt());
+        Figure[] figures = {circle,isoscelesTrapezoid,rectangle,square,rightTriangle};
```

Comment:
Such method looks difficult for reading and understanding. Try to simplify it, maybe by using methods for each figure. 
Also, try to use inner variables to describe which parameters you provide
new Rectangle(colorSuplier.getRandomColor(),random.nextInt(),
                random.nextInt()) ------> 
 new Rectangle(color, width, length);